starfish/codebook.py

```diff
@@ -513,24 +532,33 @@ def _view_row_as_element(array: np.ndarray) -> np.ndarray:
                      'formats': ncols * [array.dtype]}
             return array.view(dtype)
 
+        self._validate_decode_intensity_input_matches_codebook_shape(intensities)
+
         max_channels = intensities.argmax(Indices.CH.value)
         codes = self.argmax(Indices.CH.value)
 
+        # calculate distance scores by evaluating the fraction of signal in each round that is
+        # found in the non-maximal channels.
+        max_intensities = intensities.max(Indices.CH.value)
+        round_intensities = intensities.sum(Indices.CH.value)
+        distance = 1 - (max_intensities / round_intensities).mean(Indices.ROUND.value)
```

Comment:
this quality score is interesting -- but not an established one. perhaps leave a TODO to further explore this? and an issue